test/csuite/wt6616_checkpoint_oldest_ts/main.c

```diff
@@ -0,0 +1,372 @@
+/*-
+ * Public Domain 2014-2020 MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_util.h"
+
+#include <sys/wait.h>
+#include <signal.h>
+
+static char home[1024]; /* Program working dir */
+
+/*
+ * Spin up a child process to do operations and checkpoint. For each set of operations on a key,
+ * insert key X at timestamp X, move the stable timestamp to X, and delete the key at timestamp X+1.
+ * If the amount of data is larger than MAX_DATA, move the oldest timestamp to make half of the data
+ * obsolete.
+ *
+ * In verification, kill the child process and reopen the database to run recovery. Query the
+ * database to get the stable and oldest timestamp. For keys from oldest to stable timestamp, make
+ * sure each key X is visible at timestamp X.
+ */
+#define KEY_FORMAT ("%010" PRIu64)
+
+#define MAX_CKPT_INVL 5 /* Maximum interval between checkpoints */
+#define MAX_DATA 1000
+#define MAX_TIME 40
+#define MIN_TIME 10
+
+static const char *const uri = "table:wt6616-checkpoint-oldest-ts";
+
+static const char *const ckpt_file = "checkpoint_done";
+
+static bool log_table = false;
+
+#define ENV_CONFIG                                            \
+    "cache_size=50M,"                                         \
+    "create,"                                                 \
+    "eviction_updates_target=20,eviction_updates_trigger=90," \
+    "statistics=(fast),statistics_log=(wait=1,json=true),"    \
+    "timing_stress_for_test=[checkpoint_slow]"
+
+#define ENV_CONFIG_REC "log=(archive=false,recover=on)"
```

Comment:
I see this is based of `timestamp_abort` which is good. However I see you removed the `log=(archive=true,file_max=10M,enabled),` settings from the configuration string. The `log_table` setting and the recovery configuration are not useful if logging is not enabled when the database is created.

Reply:
fixed.